### DISCUSSION TITLE
Crtl-F WPF Instance

### DISCUSSION THREAD

**Discussion by mbowles201** [cefsharp/CefSharp#4356] [Q&A] [answered]
After a lot of searching, I found a bunch of old posts about sending a "find" command to the browser.  They all mention WinForms.  Is there a usable solution to passing the Crtl-F command to the browser to do a stander page search in a WPF application?

**Comment by amaitland** [comment, marked as answer]
> They all mention WinForms.

The fundamentals are the same for `WPF`.

1. Implement [IFindHandler](https://cefsharp.github.io/api/107.1.x/html/Methods_T_CefSharp_IFindHandler.htm)

2. Assign instance of IFindHandler
```c#
chromiumWebBrowser.FindHandler = new MyFindHandler();
```

3. Call the [Find](https://cefsharp.github.io/api/107.1.x/html/M_CefSharp_WebBrowserExtensions_Find_1.htm) method to initiate the find with the result being passed to [OnFindResult](https://cefsharp.github.io/api/107.1.x/html/M_CefSharp_IFindHandler_OnFindResult.htm).

It's up to developers to provide their own UI if required.

**Reply by mbowles201** [reply]
Understood.   Should the selectionRect have coords if the count is 1 or more?
